codec-http/src/main/java/io/netty/handler/codec/http/multipart/AbstractDiskHttpData.java

```diff
@@ -300,15 +300,17 @@ public ByteBuf getChunk(int length) throws IOException {
         }
         int read = 0;
         ByteBuffer byteBuffer = ByteBuffer.allocate(length);
-        while (read < length) {
-            int readnow = fileChannel.read(byteBuffer);
-            if (readnow == -1) {
-                fileChannel.close();
-                fileChannel = null;
-                break;
-            } else {
-                read += readnow;
+        try {
+            while (read < length) {
+                int readnow = fileChannel.read(byteBuffer);
+                if (readnow == -1) {
+                    break;
+                } else {
+                    read += readnow;
+                }
             }
+        } finally {
+                fileChannel.close();
```

Comment:
Previously, there was a `fileChannel = null;` after closing it. I believe that we still need it to be consistent in the `finally` part.

Reply:
ok